include/storage/io.hpp

```diff
@@ -32,6 +28,16 @@ class FileReader
     boost::filesystem::ifstream input_stream;
 
   public:
+    class LineWrapper : public std::string
+    {
+        friend std::istream & operator>>(std::istream &is, LineWrapper &line)
+        {
+            return std::getline(is, line);
+        }
+    };
+    auto GetLineIteratorBegin() { return std::istream_iterator<LineWrapper>(input_stream); }
+    auto GetLineIteratorEnd() { return std::istream_iterator<LineWrapper>(); }
```

Comment:
Hm can this be a istreambuf_iterator? I'm not sure about this but I _think_ these iterators work on formatted streams and streambuf works on a sequence of chars.